frontend/src/concepts/connectionTypes/fields/UriFormField.tsx

```diff
@@ -1,29 +1,71 @@
 import * as React from 'react';
-import { TextInput } from '@patternfly/react-core';
+import {
+  FormHelperText,
+  HelperText,
+  HelperTextItem,
+  TextInput,
+  ValidatedOptions,
+} from '@patternfly/react-core';
+import { ExclamationCircleIcon } from '@patternfly/react-icons';
 import { UriField } from '~/concepts/connectionTypes/types';
-import DataFormFieldGroup from '~/concepts/connectionTypes/fields/DataFormFieldGroup';
+import { FieldProps } from '~/concepts/connectionTypes/fields/types';
+import DefaultValueTextRenderer from '~/concepts/connectionTypes/fields/DefaultValueTextRenderer';
 
-type Props = {
-  field: UriField;
-  isPreview?: boolean;
-  value?: string;
-  onChange?: (value: string) => void;
+const URI_REGEX = new RegExp(
+  '(?:(?:https?|ftp|file)://|www.|ftp.)(?:([-A-Z0-9+&@#/%=~_|$?!:,.]*)|[-A-Z0-9+&@#/%=~_|$?!:,.])*(?:([-A-Z0-9+&@#/%=~_|$?!:,.]*)|[A-Z0-9+&@#/%=~_|$])',
+);
+
+const validateUrl = (url: string) => {
+  try {
+    return !url || URI_REGEX.test(url);
+  } catch (e) {
+    return false;
+  }
 };
 
-const UriFormField: React.FC<Props> = ({ field, isPreview, onChange, value }) => (
-  <DataFormFieldGroup field={field} isPreview={!!isPreview}>
-    {(id) => (
+const UriFormField: React.FC<FieldProps<UriField>> = ({
+  id,
+  field,
+  mode,
+  onChange,
+  value,
+  onValidate,
+  'data-testid': dataTestId,
+}) => {
+  const isPreview = mode === 'preview';
+
+  const isValid = validateUrl(value ?? '');
+
+  React.useEffect(() => {
+    onValidate?.(isValid);
+    // do not run when callback changes
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isValid]);
+
```

Comment:
This is updating validity as soon as the user types. I think `isValid` should be a state variable that get set on entry and then updates only `onBlur`.